src/manifolds/Euclidean.jl

```diff
@@ -129,10 +129,12 @@ log_local_metric_density(M::MetricManifold{<:Manifold,EuclideanMetric}, x) = zer
     manifold_dimension(M::Euclidean)
 
 Return the manifold dimension of the [`Euclidean`](@ref) `M`, i.e.
-the product of all array dimensions.
+the product of all array dimensions and the [`real_dimension`](@ref) of the
+underlying number system.
 """
-@generated manifold_dimension(::Euclidean{N,ℝ}) where {N} = *(N.parameters...)
-@generated manifold_dimension(::Euclidean{N,ℂ}) where {N} = 2*( *(N.parameters...) )
+@generated function manifold_dimension(::Euclidean{N,𝔽}) where {N,𝔽}
+    return *(real_dimension(𝔽), N.parameters...)
```

Comment:
I advise extracting the call to `real_dimension` outside of the generated function. Calling it here makes all method of `real_dimension` defined after this method of `manifold_dimension` potentially inaccessible.

It should be something like 
```julia
@generated _product_of_dimensions(::Euclidean{N}) where N = *(N.parameters...)
function manifold_dimension(M::Euclidean{N,𝔽}) where {N,𝔽}
    return real_dimension(𝔽) * _product_of_dimensions(M)
end
```